### PR TITLE
Webhook Notifications

### PR DESCRIPTION
**Pull request by heldtogether**
This adds the ability to parse a notification from json and populates structs of the correct type

### CODE REVIEW

**Comment by choran** [comment]
Hi @heldtogether, 
I was just taking a look at this but I wasnt clear on what you were trying to do.
Do you want to be able to subscribe to Intercom webhook notifications? e.g. like [this](https://github.com/intercom/intercom-ruby#subscriptions) in Ruby? I saw your example in the readme but if you could provide more context of example use case that would be great. Just make sure I fully understand exactly what you are trying to do.
Thanks again
Cathal

**Comment by heldtogether** [comment]
@choran this functionality assumes that you have already received a webhook notification. It decodes that notification into structs of the correct type so that you can do things with the models. Responsibility for subscribing to notifications is handled elsewhere. This just lets you get proper objects from the JSON payload.

**Comment by choran** [comment]
@heldtogether ah, ok .. I get your now. I think we have something similar in our Java SDK but it is more lightweight. I do see its usefulness, I will just take another look at it now that you have clarified it.
thanks
cathal

**Comment by choran** [comment]
Hi @heldtogether 
Just taking another look at it now and will update shortly

**Comment by choran** [comment]
@heldtogether I see that you are not covering all webhook topics, for example you are not covering leads right, I guess in that case you will just return the raw data in the notification?

**Comment by heldtogether** [comment]
@choran I think I'm covering the topics listed in the docs (https://developers.intercom.com/v2.0/reference#topics), but if any others are received then the `RawData` field is still populated with the json which can be decoded as required.

**Comment by choran** [comment]
@heldtogether the only ones I noticed were the contacts related notifications, the contacts are leads (the endpoint is still called contacts so it can be a little confusing)
But I think thats ok since like you said `RawData` would catch it anyway.
Since its new functionality i.e. we dont have webhook notifications in other SDK other than Java I just need to get one more review internally here. Should be able to merge then
Thanks again
Cathal

File: README.md [hunk 1]
```diff
@@ -454,6 +454,19 @@ convo, err := intercom.Conversations.Close("1234", &closerAdmin)
...
+### Notifications
+
+If you have received a JSON webhook notification, you may want to convert it into real Intercom object. A Notification can be created from any `io.Reader`, typically a http request:
+
+```go
+var r io.Reader
+notif, err := intercom.NewNotification(r)
+```
+
+The returned Notification will contain exactly 1 of the `Compmay`, `Conversation`, `Event`, `Tag` or `User` fields populated.
```
Comment (josler) [inline README.md]: `Compmay` -> `Company`

Comment (heldtogether) [reply README.md]: Hi @josler can you expand on your comment "the objects embedded in the notification might not be exact equivalents"? I don't quite understand what you mean.

Comment (choran) [reply README.md]: Hi @heldtogether 
I think what he was referring to was that in some cases it is not the full object returned in the notification. For example, for the conversation it is only a conversation part I think and not a whole conversation object. So maybe just add a point in the README to clarify this. And then I think we are good to merge this
Thanks again

Comment (heldtogether) [reply README.md]: @choran I've updated as advised 😃

Comment (choran) [reply README.md]: Great, I will merge it so!
Thanks again for this

**Pull request merged**